src/EtagGeneratorInterface.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace ZF\HttpCache;
+
+use Zend\Http\Request as HttpRequest;
+use Zend\Http\Response as HttpResponse;
+
+interface EtagGeneratorInterface
+{
+    /**
+     * Returns an ETag for the given response.
+     *
+     * @param HttpRequest $request
+     * @param HttpResponse $response
+     * @return string Etag
+     */
+    public function generateEtag(HttpRequest $request, HttpResponse $response);
```

Comment:
Same note as for the `DefaultETagGenerator`; let's change this method to `generate()`.